Problem Statement - Food Nutrition

Intro to Python

## Data Dictionary
1. NDB_No - Nutrition database number
2. Shrt_Desc - Short description
3. Water_(g) - water in grams per 100 grams
4. Energ_Kcal - Energy in Kcal
5. Protein_(g) - Protein
6. LipidTot(g) - Total Lipid
7. Ash_(g) - Ash
8. Carbohydrt_(g) - Carbohydrate, by difference
9. FiberTD(g) - Fiber, total dietary
10. SugarTot(g) - Total Sugars
11. Calcium_(mg) - Calcium
12. Iron_(mg) - Iron
13. Magnesium_(mg) - Magnesium
14. Phosphorus_(mg) - Phosphorus
15. Potassium_(mg) - Potassium
16. Zinc_(mg) - Zinc
17. Copper_(mg) - Copper
18. Manganese_(mg) - Manganese
19. Selenium_(æg) - Selenium
20. Sodium_(mg) - Sodium

In [1]:
#Import all the necessary modules
#Import all the necessary modules
import pandas as pd
import numpy as np
import os


# Set the working directory

In [2]:
#os.chdir('C:\\Users\\jayveer\\Desktop\\DataCamp_exercises\\GL_Mentoring\\Python')

# Import Excel file

Load the Data file into Python DataFrame using pandas read_excel method

The csv that we are going to load contains the unicode format data. We will use an additional encoding parameter which will allow pandas to load the csv. 

In [3]:
data_df = pd.read_excel("Food Nutrition.xlsx")

# View Top 10 rows

 Head function is used to view the top records. The number of records to be viewed, needs to be  given in the parenthesis.

In [4]:
data_df.head(10)

NDB_No               Shrt_Desc  Water_(g)  Energ_Kcal  Protein_(g)  \
0    1001        BUTTER,WITH SALT      15.87         717         0.85   
1    1002  BUTTER,WHIPPED,W/ SALT      16.72         718         0.49   
2    1003    BUTTER OIL,ANHYDROUS       0.24         876         0.28   
3    1004             CHEESE,BLUE      42.41         353        21.40   
4    1005            CHEESE,BRICK      41.11         371        23.24   
5    1006             CHEESE,BRIE      48.42         334        20.75   
6    1007        CHEESE,CAMEMBERT      51.80         300        19.80   
7    1008          CHEESE,CARAWAY      39.28         376        25.18   
8    1009          CHEESE,CHEDDAR      37.02         404        22.87   
9    1010         CHEESE,CHESHIRE      37.65         387        23.37   

   Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  \
0          81.11     2.11            0.06           0.0           0.06   
1          78.30     1.62            2.87           0.0           0.06   
2          99.48     0.00            0.00           0.0           0.00   
3          28.74      NaN            2.34           0.0           0.50   
4            NaN     3.18            2.79           0.0           0.51   
5          27.68     2.70            0.45           0.0           0.45   
6          24.26     3.68            0.46           0.0           0.46   
7          29.20     3.28            3.06           0.0           0.00   
8          33.31     3.71            3.09           0.0           0.48   
9          30.60     3.60            4.78           0.0           0.00   

   Calcium_(mg)  Iron_(mg)  Magnesium_(mg)  Phosphorus_(mg)  Potassium_(mg)  \
0            24       0.02               2               24              24   
1            23       0.05               1               24              41   
2             4       0.00               0                3               5   
3           528       0.31              23              387             256   
4           674       0.43              24              451             136   
5           184       0.50              20              188             152   
6           388       0.33              20              347             187   
7           673       0.64              22              490              93   
8           710       0.14              27              455              76   
9           643       0.21              21              464              95   

   Sodium_(mg)  Zinc_(mg)  Copper_mg)  Manganese_(mg)  Selenium_(µg)  
0          643       0.09       0.000           0.000            1.0  
1          583       0.05       0.010           0.001            0.0  
2            2       0.01       0.001           0.000            0.0  
3         1146       2.66       0.040           0.009           14.5  
4          560       2.60       0.024           0.012           14.5  
5          629       2.38       0.019           0.034           14.5  
6          842       2.38       0.021           0.038           14.5  
7          690       2.94       0.024           0.021           14.5  
8          653       3.64       0.030           0.027           28.5  
9          700       2.79       0.042           0.012           14.5

# View Last 20 records

Tail function is used to view the last records. The number of records to be viewed, needs to be given in the parenthesis.

In [5]:
data_df.tail(20)

NDB_No                                          Shrt_Desc  Water_(g)  \
580    3934                   BABYFOOD,CORN & SWT POTATOES,STR      82.62   
581    3935  INF FORM, ABBO NUTR, SIMIL, ALIMENT, ADVAN, RT...      87.32   
582    3936  INF FORMULA,PBM PRODUC,STO BRA,RTF (FORMERLY W...      88.10   
583    3937  INF FORM,PBM PROD,STORE BRAND,LC,NOT REC (FORM...      76.00   
584    3938           INF FORMULA,PBM PRODUCTS,STORE BRAND,PDR       2.00   
585    3939       INF FORMULA,PBM PRODUCTS,STORE BRAND,SOY,RTF      88.00   
586    3940  INF FORMU,PBM PRODU,STORE BR,SOY,LIQ CONC,NOT ...      76.00   
587    3941       INF FORMULA,PBM PRODUCTS,STORE BRAND,SOY,PDR       2.00   
588    3942  INF FORMULA, MEAD JOHNS, ENFAMI, AR LIPIL, RTF...      87.00   
589    3943  INF FORMULA, MEA JOHNSO, ENFAMI, AR LIPIL, PDR...       3.00   
590    3944    INF FORM,ABBOT NUT,SIMIL NEOSU,RTF,W/ ARA & DHA      87.00   
591    3945  INF FORMULA, ABBOTT, SIMILAC, NEOSURE, PDR, W/...       2.25   
592    3946  INF FORMULA, ABB NUTR, SIMI,SENS (LACT FRE) RT...      87.00   
593    3947  INF FORMULA, ABB NUT,SIMIL,SENS,(LACT FR),LIQ ...      75.88   
594    3948  INF FORMULA, ABB NUTR,SIMIL,SENS,(LACTO FR), P...       2.20   
595    3949  INF FORMULA, ABBOTT NUTR, SIMILAC, ADVANCE, W/...      87.79   
596    3950  INF FORMULA,ABBOTT NUTR,SIMILAC,ADVANC,W/ IRON...       2.25   
597    3951  INF FORMU,ABBO NUTR,SIMILAC,ADVAN,W/ IRON,LIQ ...      76.16   
598    3952  INF FORMULA, ABB NUTR, SIMIL, ISOMIL, ADVA W/ ...      75.81   
599    3953  INF FORMULA, ABBO NUTR,SIMIL,ISOMIL, ADVANCE W...      87.54   

     Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  \
580          68         1.26           0.28     0.61           15.23   
581          67         1.80           3.63     0.48            6.77   
582          63         1.40           3.50     0.61            6.39   
583         130         2.90           7.00     0.61           13.90   
584         524        12.00          28.00     2.00           56.00   
585          63         1.80           3.50     0.61            6.09   
586         126         3.60           7.00     1.22           12.18   
587         508        13.60          27.20     5.00           52.20   
588          68         1.71           3.49     0.33            7.53   
589         509        12.73          25.97     2.46           56.03   
590          69         1.92           3.77     0.38            6.94   
591         520        14.42          28.33     3.26           51.75   
592          68         1.48           3.74     0.75            7.40   
593         128         2.73           6.95     0.80           13.64   
594         520        11.13          28.14     3.00           55.67   
595          66         1.36           3.70     0.38            6.77   
596         522        10.89          28.87     3.26           54.73   
597         127         2.64           6.89     0.79           13.52   
598         128         3.13           6.98     0.88           13.21   
599          66         1.61           3.59     0.66            6.70   

     Fiber_TD_(g)  Sugar_Tot_(g)  Calcium_(mg)  Iron_(mg)  Magnesium_(mg)  \
580           1.8           4.00            18       0.31              10   
581           0.0           4.40            69       1.18               5   
582           0.0           6.40            41       1.18               5   
583           0.0          13.90            83       2.36               9   
584           0.0          56.00           331       9.50              36   
585           0.0           6.10            59       1.18               7   
586           0.0          12.18           118       2.36              14   
587           0.0          52.20           453       9.10              51   
588           0.0           6.44            53       1.23               5   
589           0.0          56.03           397       9.16              41   
590           0.0          

Displaying the shape of the Data Frame in which first value is giving number of Rows and second value is giving number of columns.

In [6]:
data_df.shape

(600, 20)

In [7]:
print("The number of rows are ",data_df.shape[0],"\n","The number of columns are",data_df.shape[1])

The number of rows are  600 
 The number of columns are 20


# Check for missing values

In [8]:
data_df.isnull().sum()

NDB_No             0
Shrt_Desc          0
Water_(g)          0
Energ_Kcal         0
Protein_(g)        0
Lipid_Tot_(g)      1
Ash_(g)            1
Carbohydrt_(g)     0
Fiber_TD_(g)       0
Sugar_Tot_(g)      0
Calcium_(mg)       0
Iron_(mg)          0
Magnesium_(mg)     0
Phosphorus_(mg)    0
Potassium_(mg)     0
Sodium_(mg)        0
Zinc_(mg)          0
Copper_mg)         0
Manganese_(mg)     0
Selenium_(µg)      0
dtype: int64

There are two missing values in the data set one for each variable lipid_Tot_(g) and Ash_(g). We need to drop these missing values before proceeding. We will use the dropna method to drop the missing values.

In [9]:
data_df=data_df.dropna()

In [10]:
data_df.isnull().sum()

NDB_No             0
Shrt_Desc          0
Water_(g)          0
Energ_Kcal         0
Protein_(g)        0
Lipid_Tot_(g)      0
Ash_(g)            0
Carbohydrt_(g)     0
Fiber_TD_(g)       0
Sugar_Tot_(g)      0
Calcium_(mg)       0
Iron_(mg)          0
Magnesium_(mg)     0
Phosphorus_(mg)    0
Potassium_(mg)     0
Sodium_(mg)        0
Zinc_(mg)          0
Copper_mg)         0
Manganese_(mg)     0
Selenium_(µg)      0
dtype: int64

We can now see that there are no missing values in our data set.

# Summary of the data

To see the summary of any dataframe we will use describe function.

In [11]:
data_df.describe()

NDB_No   Water_(g)  Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  \
count   598.000000  598.000000  598.000000   598.000000     598.000000   
mean   2305.448161   60.415870  186.521739     8.640385       8.849649   
std    1081.807819   31.929817  158.540534    10.939719      12.262311   
min    1001.000000    0.200000    0.000000     0.000000       0.000000   
25%    1173.250000   39.167500   63.000000     1.702500       0.697500   
50%    2052.500000   76.180000  106.000000     3.900000       3.475000   
75%    3190.750000   85.987500  314.500000    12.385000      13.000000   
max    3953.000000   99.900000  876.000000    84.080000      99.480000   

          Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  Calcium_(mg)  \
count  598.000000      598.000000    598.000000     598.000000     598.00000   
mean     2.542040       19.395920      2.587458       9.122726     262.01505   
std      6.494483       21.522166      7.594803      13.927320     376.94743   
min      0.000000        0.000000      0.000000       0.000000       0.00000   
25%      0.490000        5.560000      0.000000       0.560000      23.25000   
50%      0.890000       10.840000      0.000000       4.820000     102.00000   
75%      3.260000       19.147500      1.200000      10.692500     340.00000   
max     99.800000       86.680000     53.200000      74.460000    2240.00000   

        Iron_(mg)  Magnesium_(mg)  Phosphorus_(mg)  Potassium_(mg)  \
count  598.000000      598.000000       598.000000      598.000000   
mean     4.371622       41.760870       188.610368      311.349498   
std     12.162654       81.460504       226.962501      496.549282   
min      0.000000        0.000000         0.000000        0.000000   
25%      0.182500        8.000000        35.000000       95.000000   
50%      0.415000       13.000000       100.500000      141.000000   
75%      1.897500       33.000000       264.250000      255.000000   
max    123.600000      711.000000      1349.000000     4740.000000   

        Sodium_(mg)   Zinc_(mg)  Copper_mg)  Manganese_(mg)  Selenium_(µg)  
count    598.000000  598.000000  598.000000      598.000000     598.000000  
mean     348.270903    1.532475    0.151381        1.014246      12.426254  
std     1830.690933    1.970745    0.279639        4.220256      21.833695  
min        0.000000    0.000000    0.000000        0.000000       0.000000  
25%       21.250000    0.300000    0.026000        0.021000       1.800000  
50%       50.000000    0.670000    0.050000        0.140000       3.650000  
75%      170.750000    2.360000    0.110000        0.677000      14.500000  
max    38758.000000   17.460000    2.467000       60.127000     208.100000

# Create a vector “test” using the top 10 values of variable Protein_(g)

we will pass the variable name inside the square brackets of data_df and then use head function to retrieve the top 10 records for that variable.

In [12]:
test=data_df['Protein_(g)'].head(10)
test

0      0.85
1      0.49
2      0.28
5     20.75
6     19.80
7     25.18
8     22.87
9     23.37
10    23.76
11    11.12
Name: Protein_(g), dtype: float64

# Select the top 5 rows of initial 5 variables in a matrix format

To fetch the top 5 rows and initial variables we will use iloc and will pass the index both for rows and columns.

In [13]:
data_df.iloc[0:5, 0:5]

NDB_No               Shrt_Desc  Water_(g)  Energ_Kcal  Protein_(g)
0    1001        BUTTER,WITH SALT      15.87         717         0.85
1    1002  BUTTER,WHIPPED,W/ SALT      16.72         718         0.49
2    1003    BUTTER OIL,ANHYDROUS       0.24         876         0.28
5    1006             CHEESE,BRIE      48.42         334        20.75
6    1007        CHEESE,CAMEMBERT      51.80         300        19.80

# What is the data_type of the Sodium_(mg) variable

To check the class of any variable we will use dtype property.

In [14]:
data_df['Sodium_(mg)'].dtype

dtype('int64')

# Create a new variable “EPW” by dividing Energ_Kcal with the Water; what is the dimension of the new dataset?


In [15]:
data_df['EPW']=data_df['Energ_Kcal']/data_df['Water_(g)']

In [16]:
#dimension will be checked using the shape
data_df.shape

(598, 21)

In [17]:
data_df.head(5)

NDB_No               Shrt_Desc  Water_(g)  Energ_Kcal  Protein_(g)  \
0    1001        BUTTER,WITH SALT      15.87         717         0.85   
1    1002  BUTTER,WHIPPED,W/ SALT      16.72         718         0.49   
2    1003    BUTTER OIL,ANHYDROUS       0.24         876         0.28   
5    1006             CHEESE,BRIE      48.42         334        20.75   
6    1007        CHEESE,CAMEMBERT      51.80         300        19.80   

   Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  ...  \
0          81.11     2.11            0.06           0.0           0.06  ...   
1          78.30     1.62            2.87           0.0           0.06  ...   
2          99.48     0.00            0.00           0.0           0.00  ...   
5          27.68     2.70            0.45           0.0           0.45  ...   
6          24.26     3.68            0.46           0.0           0.46  ...   

   Iron_(mg)  Magnesium_(mg)  Phosphorus_(mg)  Potassium_(mg)  Sodium_(mg)  \
0       0.02               2               24              24          643   
1       0.05               1               24              41          583   
2       0.00               0                3               5            2   
5       0.50              20              188             152          629   
6       0.33              20              347             187          842   

   Zinc_(mg)  Copper_mg)  Manganese_(mg)  Selenium_(µg)          EPW  
0       0.09       0.000           0.000            1.0    45.179584  
1       0.05       0.010           0.001            0.0    42.942584  
2       0.01       0.001           0.000            0.0  3650.000000  
5       2.38       0.019           0.034           14.5     6.897976  
6       2.38       0.021           0.038           14.5     5.791506  

[5 rows x 21 columns]

# Create a subset of the dataset, where the Energ_Kcal is less than 500, what is the dimension of this new dataset?


We will create a new data frame by using the subset conditions from the original data_df and will use shape property for checking the dimension.

In [18]:
data_df_new=data_df[data_df['Energ_Kcal']<500]

In [19]:
data_df_new.shape

(562, 21)

Here what we got reduced rows as there were 36 rows where Energ_Kcal > 500.
(598-36=562)

# Find the top 10 products based on following

1. Higher the Energy_Kcal, higher the ranking


Here we will use sorting to find out based on higher energy values.We will use sort_values method to order the data. Since we are saying higher trhe value the rank needs to be higher we will sort the data in the descending order so that the top most value gets the higher rank.

In [20]:
data_df.sort_values(by='Energ_Kcal',ascending = False).head(10)# for descending order we chose ascending = False here

NDB_No                             Shrt_Desc  Water_(g)  Energ_Kcal  \
2      1003                  BUTTER OIL,ANHYDROUS       0.24         876   
1      1002                BUTTER,WHIPPED,W/ SALT      16.72         718   
0      1001                      BUTTER,WITH SALT      15.87         717   
131    1145                   BUTTER,WITHOUT SALT      17.94         717   
125    1137                        EGG,YOLK,DRIED       3.08         669   
122    1134  EGG,WHL,DRIED,STABILIZED,GLUCOSE RED       1.87         615   
121    1133                         EGG,WHL,DRIED       2.78         592   
63     1070              DESSERT TOPPING,POWDERED       1.47         577   
178    1210                    EGG MIX,USDA CMDTY       2.78         555   
62     1069             CREAM SUBSTITUTE,POWDERED       2.98         529   

     Protein_(g)  Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  \
2           0.28          99.48     0.00            0.00           0.0   
1           0.49          78.30     1.62            2.87           0.0   
0           0.85          81.11     2.11            0.06           0.0   
131         0.85          81.11     0.09            0.06           0.0   
125        33.63          59.13     3.51            0.66           0.0   
122        48.17          43.95     3.63            2.38           0.0   
121        48.05          43.90     4.13            1.13           0.0   
63          4.90          39.92     1.17           52.54           0.0   
178        35.60          34.50     3.15           23.97           0.0   
62          2.48          32.92     2.34           59.29           0.0   

     Sugar_Tot_(g)  ...  Iron_(mg)  Magnesium_(mg)  Phosphorus_(mg)  \
2             0.00  ...       0.00               0                3   
1             0.06  ...       0.05               1               24   
0             0.06  ...       0.02               2               24   
131           0.06  ...       0.02               2               24   
125           0.23  ...       9.56              26             1040   
122           0.00  ...       8.28              49              715   
121           0.56  ...       7.20              34              629   
63           52.54  ...       0.03               7               74   
178           2.46  ...       3.23              11              451   
62            7.77  ...       0.26               1              288   

     Potassium_(mg)  Sodium_(mg)  Zinc_(mg)  Copper_mg)  Manganese_(mg)  \
2                 5            2       0.01       0.001           0.000   
1                41          583       0.05       0.010           0.001   
0                24          643       0.09       0.000           0.000   
131              24           11       0.09       0.016           0.004   
125             264          149       7.73       0.000           0.185   
122             515          548       5.71       0.270           0.150   
121             540          476       3.15       0.203           0.058   
63              166          122       0.08       0.118           0.225   
178             373          576       2.76       0.149           0.056   
62              669          124       0.07       0.018           0.011   

     Selenium_(µg)          EPW  
2              0.0  3650.000000  
1              0.0    42.942584  
0              1.0    45.179584  
131            1.0    39.966555  
125          139.3   217.207792  
122          121.1   328.877005  
121          164.7   212.949640  
63             0.6   392.517007  
178          118.0   199.640288  
62             0.0   177.516779  

[10 rows x 21 columns]

2. Lower the water content, higher the ranking

Since we are saying lower the value the rank needs to be higher,we will sort the data in the ascending order so that the lower most value gets the higher rank.

In [21]:
data_df.sort_values(by='Water_(g)',ascending = True).head(10)# for ascending order we chose ascending = True here

NDB_No                                          Shrt_Desc  Water_(g)  \
311    2074        SEASONING MIX,DRY,SAZON,CORIANDER & ANNATTO       0.20   
295    2047                                         SALT,TABLE       0.20   
2      1003                               BUTTER OIL,ANHYDROUS       0.24   
63     1070                           DESSERT TOPPING,POWDERED       1.47   
175    1206                                 CREAM SUB,FLAV,PDR       1.52   
443    3184           BABYFOOD,CRL,WHL WHEAT,W/ APPLS,DRY FORT       1.70   
122    1134               EGG,WHL,DRIED,STABILIZED,GLUCOSE RED       1.87   
587    3941       INF FORMULA,PBM PRODUCTS,STORE BRAND,SOY,PDR       2.00   
535    3821  INF FORMULA. MEAD JOHNSON, PREGESTIMIL, W/IRON...       2.00   
584    3938           INF FORMULA,PBM PRODUCTS,STORE BRAND,PDR       2.00   

     Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  \
311           0         0.00           0.00    99.80            0.00   
295           0         0.00           0.00    99.80            0.00   
2           876         0.28          99.48     0.00            0.00   
63          577         4.90          39.92     1.17           52.54   
175         482         0.68          21.47     0.79           75.42   
443         402         6.60           4.80     3.70           83.20   
122         615        48.17          43.95     3.63            2.38   
587         508        13.60          27.20     5.00           52.20   
535         519        14.00          28.00     3.20           52.80   
584         524        12.00          28.00     2.00           56.00   

     Fiber_TD_(g)  Sugar_Tot_(g)  ...  Iron_(mg)  Magnesium_(mg)  \
311           0.0           0.00  ...       0.00             100   
295           0.0           0.00  ...       0.33               1   
2             0.0           0.00  ...       0.00               0   
63            0.0          52.54  ...       0.03               7   
175           1.2          58.01  ...       0.63              17   
443           6.7          26.66  ...      45.00             140   
122           0.0           0.00  ...       8.28              49   
587           0.0          52.20  ...       9.10              51   
535           0.0          51.00  ...       9.40              55   
584           0.0          56.00  ...       9.50              36   

     Phosphorus_(mg)  Potassium_(mg)  Sodium_(mg)  Zinc_(mg)  Copper_mg)  \
311              150              89        17000       0.88       0.008   
295                0               8        38758       0.10       0.030   
2                  3               5            2       0.01       0.001   
63                74             166          122       0.08       0.118   
175               28              90          123       0.23       0.135   
443              200             500           66       2.60       0.620   
122              715             515          548       5.71       0.270   
587              317             528          151       3.80       0.355   
535              380             550          240       5.00       0.380   
584              221             441          118       3.90       0.370   

     Manganese_(mg)  Selenium_(µg)          EPW  
311           0.677           43.4     0.000000  
295           0.100            0.1     0.000000  
2             0.000            0.0  3650.000000  
63            0.225            0.6   392.517007  
175           0.140            1.3   317.105263  
443           2.520           64.7   236.470588  
122           0.150          121.1   328.877005  
587           0.677           11.0   254.000000  
535           0.677           14.0   259.500000  
584           0.677           11.0   262.000000  

[10 rows x 21 columns]

# Create a subset of the data where product_desc contains “CHEESE” and list down the summary statistics of the subset


To find a specific value we will use the contains method. We will pass the string in the paranthesis which needs to be located in the dataframe variable.

In [22]:
data_df_cheese= data_df[data_df['Shrt_Desc'].str.contains("CHEESE")]

In [23]:
data_df_cheese.shape

(74, 21)

In [24]:
data_df_cheese.describe()

NDB_No  Water_(g)  Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  \
count    74.000000  74.000000   74.000000    74.000000      74.000000   
mean   1102.351351  48.021892  302.945946    20.956216      22.258378   
std      95.658702  14.181385   98.547797     6.423714       9.960468   
min    1006.000000  13.440000   72.000000     6.150000       0.000000   
25%    1024.250000  39.305000  264.000000    17.045000      19.480000   
50%    1042.500000  44.550000  330.000000    21.760000      25.320000   
75%    1181.750000  52.415000  372.500000    24.502500      29.800000   
max    1271.000000  82.480000  466.000000    37.860000      35.590000   

         Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  Calcium_(mg)  \
count  74.000000       74.000000     74.000000      74.000000     74.000000   
mean    4.102838        4.697568      0.005405       2.050541    632.459459   
std     1.730940        5.791562      0.028148       3.257006    296.165966   
min     1.020000        0.000000      0.000000       0.000000     53.000000   
25%     3.272500        2.010000      0.000000       0.322500    497.000000   
50%     3.790000        3.300000      0.000000       0.955000    676.500000   
75%     5.195000        5.395000      0.000000       2.365000    759.750000   
max     8.030000       42.650000      0.200000      23.670000   1375.000000   

       Iron_(mg)  Magnesium_(mg)  Phosphorus_(mg)  Potassium_(mg)  \
count  74.000000       74.000000        74.000000       74.000000   
mean    0.433784       27.243243       472.648649      157.256757   
std     0.369797       14.965541       189.169666      169.498783   
min     0.000000        5.000000       107.000000       26.000000   
25%     0.180000       20.250000       377.500000       86.000000   
50%     0.370000       27.000000       465.500000      125.000000   
75%     0.582500       29.750000       580.750000      156.500000   
max     1.900000      115.000000       931.000000     1409.000000   

       Sodium_(mg)  Zinc_(mg)  Copper_mg)  Manganese_(mg)  Selenium_(µg)  \
count    74.000000  74.000000   74.000000       74.000000      74.000000   
mean    842.094595   2.609595    0.068986        0.081419      15.709459   
std     449.944263   1.094450    0.138540        0.181131       6.324104   
min      21.000000   0.330000    0.008000        0.002000       2.800000   
25%     601.000000   1.960000    0.025250        0.011000      14.500000   
50%     707.500000   2.800000    0.031500        0.022000      14.500000   
75%    1123.750000   3.500000    0.036000        0.038750      16.925000   
max    1809.000000   4.440000    0.732000        0.677000      35.800000   

             EPW  
count  74.000000  
mean    7.536921  
std     4.872384  
min     0.872939  
25%     4.894779  
50%     7.448191  
75%     9.510973  
max    34.672619

# Using the cut function on water variable divide the whole data into 6 bins, list down the summary statistics of all the 6 bins.  


In [25]:
data_bins=data_df

In [26]:
data_bins['bins']=pd.cut(data_bins['Water_(g)'],6, labels =["A", "B", "C","E","F","G"])

In [27]:
data_bins.head(5)

NDB_No               Shrt_Desc  Water_(g)  Energ_Kcal  Protein_(g)  \
0    1001        BUTTER,WITH SALT      15.87         717         0.85   
1    1002  BUTTER,WHIPPED,W/ SALT      16.72         718         0.49   
2    1003    BUTTER OIL,ANHYDROUS       0.24         876         0.28   
5    1006             CHEESE,BRIE      48.42         334        20.75   
6    1007        CHEESE,CAMEMBERT      51.80         300        19.80   

   Lipid_Tot_(g)  Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  ...  \
0          81.11     2.11            0.06           0.0           0.06  ...   
1          78.30     1.62            2.87           0.0           0.06  ...   
2          99.48     0.00            0.00           0.0           0.00  ...   
5          27.68     2.70            0.45           0.0           0.45  ...   
6          24.26     3.68            0.46           0.0           0.46  ...   

   Magnesium_(mg)  Phosphorus_(mg)  Potassium_(mg)  Sodium_(mg)  Zinc_(mg)  \
0               2               24              24          643       0.09   
1               1               24              41          583       0.05   
2               0                3               5            2       0.01   
5              20              188             152          629       2.38   
6              20              347             187          842       2.38   

   Copper_mg)  Manganese_(mg)  Selenium_(µg)          EPW  bins  
0       0.000           0.000            1.0    45.179584     A  
1       0.010           0.001            0.0    42.942584     A  
2       0.001           0.000            0.0  3650.000000     A  
5       0.019           0.034           14.5     6.897976     C  
6       0.021           0.038           14.5     5.791506     E  

[5 rows x 22 columns]

In [28]:
data_bins["bins"].value_counts()

G    193
F    175
A    127
C     52
E     41
B     10
Name: bins, dtype: int64

we can either do this in one go, like shown below

In [29]:
data_bins.groupby("bins").describe().T

bins                    A            B            C            E            F  \
NDB_No count   127.000000    10.000000    52.000000    41.000000   175.000000   
       mean   2427.000000  1291.200000  1127.615385  1225.097561  2379.885714   
       std    1006.346851   610.900938   103.775239   250.416234  1123.918049   
       min    1001.000000  1023.000000  1006.000000  1007.000000  1012.000000   
       25%    2003.500000  1034.250000  1029.750000  1072.000000  1183.000000   
...                   ...          ...          ...          ...          ...   
EPW    min       0.000000    10.971787     4.155412     1.551221     0.721154   
       25%      36.045815    12.462685     6.718986     2.846975     0.979906   
       50%      72.244898    14.511953     7.913460     4.373033     1.230272   
       75%     165.333333    17.114903     9.482712     5.405937     1.685755   
       max    3650.000000    39.966555    11.341632     6.651463     2.946455   

bins                    G  
NDB_No count   193.000000  
       mean   2757.367876  
       std     947.352355  
       min    1059.000000  
       25%    2053.000000  
...                   ...  
EPW    min       0.000000  
       25%       0.508475  
       50%       0.638468  
       75%       0.726846  
       max       1.163209  

[160 rows x 6 columns]

or we can create subsets of the data on the bases of bins

In [30]:
A=data_bins[data_bins["bins"]=="A"]
B=data_bins[data_bins["bins"]=="B"]
C=data_bins[data_bins["bins"]=="C"]
E=data_bins[data_bins["bins"]=="E"]
F=data_bins[data_bins["bins"]=="F"]
G=data_bins[data_bins["bins"]=="G"]

In [31]:
A.describe()# You can do for others

NDB_No   Water_(g)  Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  \
count   127.000000  127.000000  127.000000   127.000000     127.000000   
mean   2427.000000    5.964094  405.181102    16.981339      16.340157   
std    1006.346851    3.634829  123.142286    15.983079      16.559315   
min    1001.000000    0.200000    0.000000     0.000000       0.000000   
25%    2003.500000    2.990000  324.000000     9.055000       4.320000   
50%    2035.000000    5.440000  391.000000    12.010000      12.600000   
75%    3214.500000    8.460000  505.500000    18.030000      26.855000   
max    3950.000000   16.720000  876.000000    84.080000      99.480000   

          Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  Calcium_(mg)  \
count  127.000000      127.000000    127.000000     127.000000    127.000000   
mean     7.032441       53.572756      9.881102      20.913150    618.330709   
std     12.780734       21.142774     14.068132      23.569114    530.134186   
min      0.000000        0.000000      0.000000       0.000000      0.000000   
25%      2.925000       50.360000      0.000000       0.395000    236.500000   
50%      4.000000       55.830000      1.400000       7.270000    470.000000   
75%      7.010000       68.540000     14.700000      51.000000    912.000000   
max     99.800000       86.680000     53.200000      74.460000   2240.000000   

        Iron_(mg)  Magnesium_(mg)  Phosphorus_(mg)  Potassium_(mg)  \
count  127.000000      127.000000       127.000000      127.000000   
mean    17.172126      136.440945       358.480315      935.023622   
std     21.835832      137.197902       263.466297      801.534456   
min      0.000000        0.000000         0.000000        5.000000   
25%      3.195000       41.000000       169.500000      500.500000   
50%      9.260000       90.000000       299.000000      636.000000   
75%     20.470000      180.500000       449.000000     1263.500000   
max    123.600000      711.000000      1349.000000     4740.000000   

        Sodium_(mg)   Zinc_(mg)  Copper_mg)  Manganese_(mg)  Selenium_(µg)  \
count    127.000000  127.000000  127.000000      127.000000     127.000000   
mean     858.181102    3.765433    0.485024        3.675102      26.554331   
std     3848.331336    2.743998    0.452203        8.656279      37.762413   
min        0.000000    0.010000    0.000000        0.000000       0.000000   
25%       51.000000    1.795000    0.138500        0.124500       6.750000   
50%      151.000000    3.860000    0.386000        0.677000      14.100000   
75%      336.500000    5.145000    0.639000        2.660000      27.300000   
max    38758.000000   17.460000    2.467000       60.127000     208.100000   

               EPW  
count   127.000000  
mean    131.393840  
std     325.416276  
min       0.000000  
25%      36.045815  
50%      72.244898  
75%     165.333333  
max    3650.000000

OR make loop for it.

In [40]:
X=[A,B,C,E,F,G]
for i in X:
    print(i.describe())

            NDB_No   Water_(g)  Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  \
count   127.000000  127.000000  127.000000   127.000000     127.000000   
mean   2427.000000    5.964094  405.181102    16.981339      16.340157   
std    1006.346851    3.634829  123.142286    15.983079      16.559315   
min    1001.000000    0.200000    0.000000     0.000000       0.000000   
25%    2003.500000    2.990000  324.000000     9.055000       4.320000   
50%    2035.000000    5.440000  391.000000    12.010000      12.600000   
75%    3214.500000    8.460000  505.500000    18.030000      26.855000   
max    3950.000000   16.720000  876.000000    84.080000      99.480000   

          Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  Calcium_(mg)  \
count  127.000000      127.000000    127.000000     127.000000    127.000000   
mean     7.032441       53.572756      9.881102      20.913150    618.330709   
std     12.780734       21.142774     14.068132      23.569114    530.134186   
min      0.00

            NDB_No   Water_(g)  Energ_Kcal  Protein_(g)  Lipid_Tot_(g)  \
count   175.000000  175.000000  175.000000   175.000000     175.000000   
mean   2379.885714   77.783600  105.457143     5.172114       4.262629   
std    1123.918049    3.955846   34.991413     4.345693       4.625422   
min    1012.000000   66.860000   60.000000     0.100000       0.000000   
25%    1183.000000   75.045000   78.000000     2.080000       0.300000   
50%    3011.000000   79.000000   98.000000     3.600000       2.700000   
75%    3227.500000   81.240000  128.000000     8.085000       6.970000   
max    3952.000000   83.200000  208.000000    15.690000      19.520000   

          Ash_(g)  Carbohydrt_(g)  Fiber_TD_(g)  Sugar_Tot_(g)  Calcium_(mg)  \
count  175.000000      175.000000    175.000000     175.000000    175.000000   
mean     0.879371       11.888057      0.498286       6.842343     94.822857   
std      0.857238        6.400474      1.338659       6.095994     79.846674   
min      0.14